imap_processing/cdf/utils.py

```diff
@@ -128,6 +129,8 @@ def write_cdf(dataset: xr.Dataset):
     # Insert the final attribute:
     # The Logical_file_id is always the name of the file without the extension
     dataset.attrs["Logical_file_id"] = file_path.stem
+    # Add the processing version to the dataset attributes
+    dataset.attrs["imap_processing_version"] = imap_processing.__version__
```

Comment:
Open to other naming suggestions here for that dataset level attribute.

Reply:
GLOWS requested "ground_software_version", so maybe along those lines?

Reply:
:+1: That would be fine with me too. The only issue I'd see is if we want to put the versions of multiple software packages being used in, like do we want to start including `imap-data-access-version`, or something else informative too?

Reply:
Probably I would prefer "ground_software_version", since presumably updating the `imap-data-access` dependency would result in a new `ground_software_version`. I'd prefer it to be super obvious to people what the main version they should care about is, and we can add more specific info later/if needed.

Reply:
I would actually prefer to call it `imap_processing_version` then because that is super explicit that is this package's version we are referring to. IMO `ground_software_version` is more encompassing and could mean container hash, imap_processing version hash, something else we manually set/stamp ourselves as this is a version.

@maxinelasp I see that you put `ground_software_version` in GLOWS. Would this mess things up if we also injected an `imap_processing_version` (or overwrite the `ground_software_version` metadata you're already declaring) here?

Reply:
But it's fine with me

Reply:
`ground_software_version` was voted the winner.

@maxinelasp, you put in GLOWS metadata that it is of the format `vXXX`, with the dynamic versioning it will actually be whatever the current tag is: `v0.1.2` or if you're in development it will show you commit info as well: `0.2.0.post173.dev0+cde8921` Is that OK for your situation, or do we need to update the GLOWS documentation/attributes to account for this?

Reply:
Nope, that should work, and I probably just got confused about which version I was using there. I would expect it to be the semantic version of the repo.